packages/flutter/lib/src/material/icon.dart

```diff
@@ -65,13 +83,13 @@ class Icon extends StatelessComponent {
       category = parts[0];
       subtype = parts[1];
     }
-    // TODO(eseidel): This clearly isn't correct.  Not sure what would be.
-    // Should we use the ios images on ios?
-    String density = 'drawable-xxhdpi';
+    // TODO(kgiesing): Should selector logic be moved down a level so that
+    // any generic asset can be resolved using consistent contextual logic?
```

Comment:
The way we've been approaching this kind of thing is to have layers of opinion. Bottom layer: doesn't care about DPI at all. Next layer: general map from breakpoints to specific images. Next layer: a predefined set of breakpoints and a scheme for file names. Next layer: you just say the icon, and we figure out which file name to use. Each layer is built on the previous one, and you can hook in wherever you need to.